java/java-hello-world/src/main/java/cloudcode/helloworld/HelloWorldApplication.java

```diff
@@ -9,8 +9,12 @@
 @EnableAutoConfiguration
 @ComponentScan
 public class HelloWorldApplication {
-	public static void main(String[] args) throws Exception {
-		SpringApplication.run(HelloWorldApplication.class, args);
-	}
-
+    public static void main(final String[] args) throws Exception {
+        String value = System.getenv("PORT");
```

Comment:
typically if it's not set, we set it to 8080 -- may not be appropriate in your world.

Reply:
We decided in a Kubernetes context it's better to fail fast if expected environment variables aren't passed in properly